src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs

```diff
@@ -101,12 +101,13 @@ public long RefreshIn
 
         internal static MsalTokenResponse CreateFromBrokerResponse(Dictionary<string, string> responseDictionary)
         {
-            if (responseDictionary.ContainsKey(BrokerResponseConst.ErrorMetadata))
+            if (responseDictionary.ContainsKey(BrokerResponseConst.BrokerErrorCode) ||
```

Comment:
This is a not negligible amount of logic that should be unit tested (it's not ios code)

Reply:
once i can test again, i can pull some actual decoded broker responses and write some tests. 

---
In reply to: [323189847](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1383#discussion_r323189847) [](ancestors = 323189847)